Relying Party Remote Presentation alignments with standards (#400)
italia/eidas-it-wallet-docs: docs/en/remote-flow.rst

  * - **13**
    - When the Wallet Instance capabilities discovery is not supported by RP, the Wallet Instance request the signed Request Object using the HTTP method GET.
  * - **14**

>     - The RP issues the Request Object signin it using one of its cryptographic private keys, where their public parts have been published within its Entity Configuration (`metadata.openid_wallet_relying_party.jwks`). The Wallet Instance obtains the signed Request Object.

Reviewer: Secondo le specs correnti, i metadata dell'RP sono in `metadata.wallet_relying_party.jwks`. non, `metadata.openid_wallet_relying_party.jwks` (si osservi l'omissione del prefisso `openid_`). Quale fa fede?

Author: decisamente wallet_relying_party

>     - The RP issues the Request Object signin it using one of its cryptographic private keys, where their public parts have been published within its Entity Configuration (`metadata.openid_wallet_relying_party.jwks`). The Wallet Instance obtains the signed Request Object.

Author: ```suggestion
    - The RP issues the Request Object signin it using one of its cryptographic private keys, where their public parts have been published within its Entity Configuration (`metadata.wallet_relying_party.jwks`). The Wallet Instance obtains the signed Request Object.
```

  Content-Type: application/json

  {

>       "authorization_endpoint": "openid4vp:",

Author: ```suggestion
      "authorization_endpoint": "openid4vp://",
```

  - ``presentation_definition``: JSON object according to `Presentation Exchange <https://identity.foundation/presentation-exchange/spec/v2.0.0/>`_. This parameter MUST not be present when ``presentation_definition_uri`` or ``scope`` are present.
  - ``presentation_definition_uri``: Not supported. String containing an HTTPS URL pointing to a resource where a Presentation Definition JSON object can be retrieved. This parameter MUST be present when ``presentation_definition`` parameter or a ``scope`` value representing a Presentation Definition is not present. 

>   - ``client_metadata``: A JSON object containing the Relying Party metadata values. If the ``client_metadata`` parameter is present when ``client_id_scheme`` is ``entity_id``, the client_metadata MUST only contains the member ``jwks`` and containing within it ephemeral cryptographic public keys to be used only for the Wallet Instance with which the Relying Party is interacting with.

Author: not sure to want to keep this.

even if there are claims about the importance of using ephemeral keys by RP, there might be other concerns about using jwks outside of the trust chain, that would allow an adversary to hijack the response and therefore be able to decrypt it.

further conversation are required, even within the DCP WG  and with the italian team as well.

Reviewer: To me, the purpose of this change is slightly unclear. 
Does it imply that `client_metadata` in Request object goes from "MUST NOT be present" to "OPTIONAL: when used it must contain an ephemeral JWKS"?
If so, where did the necessity of ephemeral key arise from?

Author: from Google Credential API, something that will be formally included later on

at the current stage I'd prefer and suggest to move this out from this PR and handle it in a separate issue

from my perspective even using credential api it is still unclear this requirement.

Reviewer: Ah, I see
I'll try to learn more about Google Credential API in the near future as I don't know anything about it.

Author: it is now included within the openid4vp specs, annex, in the editor's copy on github

Author: @Zicchio I have moved it out, to this issue: https://github.com/italia/eudi-wallet-it-docs/issues/408

>   - ``client_metadata``: A JSON object containing the Relying Party metadata values. If the ``client_metadata`` parameter is present when ``client_id_scheme`` is ``entity_id``, the client_metadata MUST only contains the member ``jwks`` and containing within it ephemeral cryptographic public keys to be used only for the Wallet Instance with which the Relying Party is interacting with.

Author: ```suggestion
  - ``client_metadata``: A JSON object containing the Relying Party metadata values. If the ``client_metadata`` parameter is present when ``client_id_scheme`` is ``entity_id``, the ``client_metadata`` MUST only contain the member ``jwks`` and containing ephemeral cryptographic public keys to be used only for the Wallet Instance with which the Relying Party is interacting with for this specific session.
```

      - The requested Digital Credential (one or more, in format of SD-JWT VC or MDOC CBOR)
      - The Wallet Attestation
  * - **presentation_submission**

>     - JSON Object containing the mappings between the requested Verifiable Credentials and where to find them within the returned Verifiable Presentation Token, according to the `Presentation Exchange <https://identity.foundation/presentation-exchange/spec/v2.0.0/>`_. This is expressed via elements in the ``descriptor_map`` array (Input Descriptor Mapping Objects) that contain a field called ``path``, which MUST have the value $ (top level root path) when only one Verifiable Presentation is contained in the VP Token, and MUST have the value $[n] (indexed path from root) when there are multiple Verifiable Presentations, where ``n`` is the index to select. The Relyingh PArty receiving the `presentation_submission` descriptor therefore is able to use the corred method to decode each credential data format provided withi nthe ``vp_token``.

Author: ```suggestion
    - JSON Object containing the mappings between the requested Verifiable Credentials and where to find them within the returned Verifiable Presentation Token, according to the `Presentation Exchange <https://identity.foundation/presentation-exchange/spec/v2.0.0/>`_. This is expressed via elements in the ``descriptor_map`` array (Input Descriptor Mapping Objects) that contain a field called ``path``, which MUST have the value $ (top level root path) when only one Verifiable Presentation is contained in the VP Token, and MUST have the value $[n] (indexed path from root) when there are multiple Verifiable Presentations, where ``n`` is the index to select. The Relying Party receiving the `presentation_submission` descriptor therefore is able to use the correct method to decode each credential data format provided within the ``vp_token``.
```
